lib/client.js

```diff
@@ -92,7 +92,7 @@ class Metacom extends EventEmitter {
         const target = interfaceName + '/' + methodName;
         const packet = { call: callId, [target]: args };
         const dest = new URL(this.url);
-        const protocol = dest.protocol === 'ws:' ? 'http' : 'https';
+        const protocol = dest.protocol.endsWith('s') ? 'https' : 'http';
```

Comment:
@tshemsedinov both `ws` and `wss` end with `s` that's why I used the check I used.